src/lib/OS/MemRegion.cc

```diff
@@ -238,29 +223,18 @@ int64_t MemRegion::mmapRegion(uint64_t startAddr, uint64_t length, int prot,
           << std::endl;
       return -1;
     }
-
     if (!((startAddr >= mmapStart_) && (startAddr + size < mmapEnd_))) {
       std::cout << "[SimEng:MemRegion] Provided address range doesn't exist in "
                    "the mmap range: "
                 << startAddr << " - " << startAddr + size << std::endl;
       return -1;
     }
-    mapped = isVmMapped(startAddr, size);
-  }
-
-  // if not fixed and hint is provided then we need to check if the hint
-  // address is available. If not we allocate vma at the most optimal address
-  // available.
-  VMA* vma = new VMA(prot, flags, size, hfmmap);
-  uint64_t returnAddress = 0;
-  if (startAddr && !mapped) {
     returnAddress = addVma(vma, startAddr);
   } else {
     // TODO: Check if offset should be contained in HostBackedFileMMap,
     // because hfmmaps are shared during unmaps.
     returnAddress = addVma(vma);
   }
```

Comment:
Is this else still needed? Given in the new `addVma` function there are checks if `startAddr` is 0 or not

Reply:
please take a look